### PR TITLE
Map

### PR DESCRIPTION
**Pull request by nyananu**
Screen Shot 2022-08-02 at 5 32 30 AM
1) Click on a marker on the map.
2) If you hover on the marker you will see where you are
3) Click on update location button
4) Location will be updated on the profile as well as my items etc. 

To test this out, please pass in the API key at `frontend/src/components/profile/Location.jsx`  line 88. (I will share the API key on slack). I tried putting it in the env. folder but somehow it's messing up my map so will look into it further

### CODE REVIEW

**Comment by imogenem** [comment]
I'm trying to run this, but I can't click on the pin or update location so far. I thiiink it might be a package issue, I was trying to solve the merge conflicts and install.

To solve your merge conflicts, it might be easiest to delete the node modules and package-lock.json in the root, and run npm install again. It looks like your package-lock.json has react-calendar in it, which was moved from the root to the frontend folder only.

**Comment by nyananu** [comment]
Ok, so after spending last night on the warnings I was getting, I have decided to just go ahead with the current functionality of the map. Meaning that this PR is no longer a WIP, it is good to go. The reasons for this are as follows:

**Warning 1:** 

> Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See [https://reactjs.org/link/unsafe-component-lifecycles](https://reactjs.org/link/unsafe-component-lifecycles) for details.
> 

> Move data fetching code or side effects to componentDidUpdate.
> 

> If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: [https://reactjs.org/link/derived-state](https://reactjs.org/link/derived-state)
> 

> Please update the following components: Wrapper
> 

There are 3 libraries to implement Google Maps in a react app, those are :

- google-map-react  (most updated)
- google-maps-react  (lazy-loading)
- react-google-maps  (HOC/extra difficult features)

I used the 2nd one, google-maps-react. The current version of this library has a problem with React.StrictMode, which is why that warning shows up, according to this [post](https://stackoverflow.com/questions/65026280/warning-using-unsafe-componentwillreceiveprops-in-strict-mode). React.StrictMode doesn’t impact production, according to this[ post](https://reactjs.org/docs/strict-mode.html). So we are good.

**Warning 2:**

> Warning: Can't call setState on a component that is not yet mounted. This is a no-op, but it might indicate a bug in your application. Instead, assign to `this.state` directly or define a `state = {};` class property with the desired state in the Wrapper component.
> 

This is also a similar warning to do with the Wrapper component of the google-maps-react library.

Apart from that, the env. API key has also been fixed, so you all can review this.

**Review by imogenem — COMMENTED** [review]
This works for me, Marketplace works as normal and I can change my location - but I also see this. Is that expected?
Screen Shot 2022-08-03 at 4 43 51 PM

**Comment by nyananu** [comment]
@imogenem  those warnings will be shown, it's ok. But as far as the map is concerned, did you put in the API key? Because the map should be showing up normally.

File: package.json [hunk 1]
```diff
@@ -12,6 +12,8 @@
     "dotenv": "^16.0.1",
     "express": "^4.18.1",
     "express-session": "^1.17.3",
+    "google-map-react": "^2.2.0",
+    "google-maps-react": "^2.0.6",
```
Comment (phreid) [inline package.json:16]: these should be in the package.json in the frontend directory, not the root directory. but that causes a problem - it seems like `google-maps-react` doesn't support React 18, which is what we have. we don't want to downgrade React. can you look into this?

Comment (phreid) [reply package.json:16]: this package supports React 18: https://www.npmjs.com/package/@react-google-maps/api

Comment (nyananu) [reply package.json:16]: ok, ill do this

**Comment by nyananu** [comment]
Closing this PR because this is not the map library we want to use.

**Pull request closed without merging**